features/step_definitions/user.rb

```diff
@@ -72,3 +72,51 @@
 Then(/^(.*) should not be able to sign up$/) do |username|
   expect(user_named(username).persisted?).to be_falsey
 end
+
+
+When(/^(.+) visits the password recovery page$/) do |username|
+  visit new_user_password_path
+end
+
+And(/^(.+) inserts his email address$/) do |username|
+  user = user_named(username)
+  fill_in :user_email, with: user.email
+end
+
+And(/^submits$/) do
+  click_button I18n.t('devise.registrations.reset_password')
+end
+
+Then(/^(\w+) gets an Email with a recovery link$/) do |username|
+  user = user_named(username)
+  reset_password_mail = ActionMailer::Base.deliveries.last
+  expect(reset_password_mail.subject).to have_content('asswor') # (P|p)asswor(d|t)
+  expect(reset_password_mail.to[0]).to eq(user.email)
+  expect(reset_password_mail.body.to_s).to have_css('a')
+end
+
+When(/^(\w+) clicks the recovery link$/) do |username|
+  reset_password_mail = ActionMailer::Base.deliveries.last
+  path_regex = /(?:"https?\:\/\/.*?)(\/.*?)(?:")/
+  path = reset_password_mail.body.match(path_regex)[1]
```

Comment:
Try to use something like capybara for webpages here as well, one example might be this gem: https://github.com/DavyJonesLocker/capybara-email

Reply:
I tried and failed. Is this line of code so bad?